File: src/test/java/com/pmd/dndplatform/CiProofTest.java
package com.pmd.dndplatform;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

/*
 * CiProofTest.java
 *
 * Phase 2 Story 4.8. TEMPORARY. DELETE THIS FILE.
 *
 * This test fails on purpose. It exists for one push, to prove that a failing
 * backend test turns the "Backend tests" check red on the pull request. Two
 * acceptance criteria on the card ask for exactly that, and a check nobody has
 * ever seen fail is only a belief that it works.
 *
 * No Spring, no database. A plain unit test is enough, because what is being
 * proved is that the check reports a failure, not what caused it.
 *
 * It must be deleted before this branch is merged. If you are reading this on
 * main, something went wrong.
 */
class CiProofTest {

    @Test
    void deliberateFailure_provesTheBackendCheckGoesRed() {
        assertThat(1).isEqualTo(2);
    }
}

File: frontend/src/CiProof.test.ts
/*
 * CiProof.test.ts
 *
 * Phase 2 Story 4.8. TEMPORARY. DELETE THIS FILE.
 *
 * The frontend half of the same proof as CiProofTest.java. It fails on purpose,
 * for one push, to show that a failing frontend test turns the "Frontend tests"
 * check red on the pull request.
 *
 * describe, it and expect are imported explicitly because the Vitest settings in
 * vite.config.ts do not switch on "globals". Without these imports the file
 * would fail with a reference error, which would still turn the check red but
 * for the wrong reason, and would prove nothing about a failing assertion.
 *
 * It must be deleted before this branch is merged.
 */
import { describe, it, expect } from 'vitest'

describe('CI proof', () => {
  it('fails on purpose, to prove the frontend check goes red', () => {
    expect(1).toBe(2)
  })
})
